src/sage/knots/link.py

```diff
@@ -4282,31 +4376,28 @@ def answer_unori(S):
             if not mirror_version:
                 return S
 
-            mirrored = [answer(L)[1] for L in S]
-            if all(mirrored):
+            sym_mut = [answer(L)[1] for L in S]
+            if all(i is SymmetryMutant.mirror_image  for i in sym_mut):
                 # all matching links are mirrored to self
-                return S, True
-            if any(i == '?' for i in mirrored):
+                return S, SymmetryMutant.mirror_image
+            if all(i is SymmetryMutant.itself  for i in sym_mut):
+                # all matching links are self itself
+                return S, SymmetryMutant.itself
+            if any(i is SymmetryMutant.unknown for i in sym_mut):
                 # unknown chirality for a matching link
-                return S, '?'
-            if any(i is None for i in mirrored):
-                # an amphicheiral link matches
-                return S, None
-            if not any(mirrored):
-                # no matching link is mirrored to self
-                return S, False
-            # finally both mirror types match
-            return S, None
+                return S, SymmetryMutant.unknown
+            # finally several mirror types match
+            return S, SymmetryMutant.mixed
 
         def answer_list(l):
             r"""
             Return a list of items of the KnotInfo database according to the keyword
             argument ``unique``.
             """
             if not unique:
-                return sorted([answer(L) for L in l])
+                return sorted(set([answer(L) for L in l]))
```

Comment:
This seems to be making the output have unique instances. Perhaps I am misunderstanding something here?

Reply:
The list `l` is merged from the individual lists of different symmetry mutant:

```
        l = []
        if proved and unique and self.is_knot():
            for k in match_lists.keys():
                if proves[k]:
                    l += match_lists[k]
        else:
            # for multi-component links there could regularily be more than one
            # matching entry
            for k in match_lists.keys():
                l += match_lists[k]
```

Therefore, for amphicheiral or reversible links there might be duplicates in it.

Reply:
Ah, I see. I misunderstood what the `unique` parameter's purpose is.